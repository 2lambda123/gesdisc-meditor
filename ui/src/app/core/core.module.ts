import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
