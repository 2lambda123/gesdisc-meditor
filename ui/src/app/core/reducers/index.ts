import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
