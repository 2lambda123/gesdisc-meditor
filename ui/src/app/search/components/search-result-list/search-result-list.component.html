<med-search-result *ngFor="let result of results | paginate: { itemsPerPage: 9, currentPage: p }" [result]="result" [model]="model" (loadDocument)="loadDocument.emit($event)"></med-search-result>
<pagination-controls (pageChange)="p = $event"></pagination-controls>