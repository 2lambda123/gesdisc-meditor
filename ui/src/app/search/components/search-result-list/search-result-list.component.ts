import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core'
import { DocCatalogEntry } from '../../../service/model/docCatalogEntry'
import { ModelCatalogEntry } from '../../../service/model/modelCatalogEntry'

@Component({
    selector: 'med-search-result-list',
    templateUrl: './search-result-list.component.html',
    styleUrls: ['./search-result-list.component.css'],
})
export class SearchResultListComponent implements OnInit {
    @Input() results: DocCatalogEntry[]
    @Input() model: ModelCatalogEntry

    @Output() loadDocument = new EventEmitter<string>()

    p: number = 1

    constructor() {}

    ngOnInit() {}
}
