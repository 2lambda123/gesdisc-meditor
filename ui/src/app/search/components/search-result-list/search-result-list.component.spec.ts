import { ComponentFixture, TestBed  } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { SearchResultListComponent } from './search-result-list.component';

import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('SearchResultListComponent', () => {
	let fixture: ComponentFixture<SearchResultListComponent>;
	let instance: SearchResultListComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
			],
			declarations: [
				SearchResultListComponent
			],
			schemas: [ NO_ERRORS_SCHEMA ]
		});

		fixture = TestBed.createComponent(SearchResultListComponent);
		instance = fixture.componentInstance;

	});

	describe('with no data', () => {
		it('should not have any initial input', () => {
			expect(instance.results).toBeUndefined();
		});
	});

	describe('with mock data', () => {

		it('should compile with minimum input (have 2 med-search-result elements)', () => {
			instance.results = [
				{ title: 'test title',
					'x-meditor': {
						modifiedOn: '2018-05-04T19:09:05.366Z',
						modifiedBy: 'test author'
					}
				},
				{ title: 'test title2',
					'x-meditor': {
						modifiedOn: '2018-05-06T19:09:05.366Z',
						modifiedBy: 'test author2'
					}
				}
			]
			fixture.detectChanges();
			expect(fixture).toMatchSnapshot();
		});
	});
});
