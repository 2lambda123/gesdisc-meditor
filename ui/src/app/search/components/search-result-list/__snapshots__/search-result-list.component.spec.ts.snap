// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultListComponent with mock data should compile with minimum input (have 2 <med-search-result /> elements in snapshot) 1`] = `
<med-search-result-list
  loadDocument={[Function EventEmitter]}
  p={[Function Number]}
  results={[Function Array]}
>
  <med-search-result /><med-search-result /><pagination-controls>
    <pagination-template
      ng-reflect-max-size="7"
    >
      
      <ul
        aria-label="Pagination"
        class="ngx-pagination"
        role="navigation"
      >
        
        <li
          class="pagination-previous disabled"
        >
          
          
          <span>
             Previous 
            <span
              class="show-for-sr"
            >
              page
            </span>
          </span>
        </li>
        <li
          class="small-screen"
        >
           1 / 1 
        </li>
        
        <li
          class="current"
        >
          
          
          
          <span
            class="show-for-sr"
          >
            You're on page 
          </span>
          <span>
            1
          </span>
        </li>
        
        <li
          class="pagination-next disabled"
        >
          
          
          <span>
             Next 
            <span
              class="show-for-sr"
            >
              page
            </span>
          </span>
        </li>
      </ul>
    </pagination-template>
  </pagination-controls>
</med-search-result-list>
`;
